ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -618,14 +819,16 @@ private void verifyState(boolean isLogSegmented) throws IOException, StoreExcept
       TreeMap<MockId, IndexValue> referenceIndexSegment = referenceIndexEntry.getValue();
       IndexSegment realIndexSegment = realIndexEntry.getValue();
       List<MessageInfo> messageInfos = new ArrayList<>();
-      FindEntriesCondition condition = new FindEntriesCondition(Long.MAX_VALUE);
       assertTrue("There should have been entries returned from the index segment",
-          realIndexSegment.getEntriesSince(null, condition, messageInfos, new AtomicLong(0)));
+          realIndexSegment.getEntriesSince(null, new FindEntriesCondition(Long.MAX_VALUE), messageInfos,
+              new AtomicLong(0)));
       assertEquals("Size of index segment differs from expected", referenceIndexSegment.size(), messageInfos.size());
       for (Map.Entry<MockId, IndexValue> referenceIndexSegmentEntry : referenceIndexSegment.entrySet()) {
         IndexValue value = realIndexSegment.find(referenceIndexSegmentEntry.getKey());
         IndexValue referenceValue = referenceIndexSegmentEntry.getValue();
-        assertEquals("Value from IndexSegment does not match expected", referenceValue.getBytes(), value.getBytes());
+        if (index.hardDeleter.enabled.get() && !deletedKeys.contains(referenceIndexSegmentEntry.getKey())) {
```

Comment:
It has to do with the fact that `HardDeleter` advances time by itself. 

If you take a look at `addDeleteEntry()`, it generates an `IndexValue` on its own (because `PersistentIndex::markAsDeleted()` does not accept or return an `IndexValue`). Because `HardDeleter` is constantly updating time, there is a difference b/w the operation time generated in `addDeleteEntry()` and the operation time generated in `PersistentIndex::markAsDeleted()`. 

I have added individual tests for all the fields also - it compares everything except the op time outside the `if` block.